beaker/application.py

```diff
@@ -311,16 +366,25 @@ def compile(self, client: Optional[AlgodClient] = None) -> tuple[str, str]:
                 overriding_name=method.name(),
             )
 
-        # Compile approval and clear programs
-        (
-            self.approval_program,
-            self.clear_program,
-            self.contract,
-        ) = self.router.compile_program(
+        # RouterResults
+        res = self.router.compile(
             version=self.teal_version,
             assemble_constants=True,
             optimize=self.optimize_options,
+            approval_filename=ARTIFACTS.approval(),
+            clear_filename=ARTIFACTS.clear(),
+            with_sourcemaps=True,
+            pcs_in_sourcemap=bool(client),
```

Comment:
TODO: should we just hard code this to "false" and therefore let beaker deal exclusively with PC source-mapping?
* If the answer is "yes" then the `pcs_in_sourcempa` param should be removed
* If the answer is "no" (also allow PyTeal PC-source-mapping), then keep this line but then we'll need to decide whether to save the pc-sourcemap artifact coming back